lib/cheers_leader.rb

```diff
@@ -1,21 +1,31 @@
 class CheerLeaderList
+  attr_reader :list
+  
   def initialize(list)
+    raise ArgumentError if !list.is_a?(Array) || list.any? {|x| x.any? { |i| i.nil? } }
     @list = list
+    sort_by_rank # Why this not WORKING ?? 
```

Comment:
this is because `list.sort_by` returns a new list, but this `initialize` method doesn't return a value (it returns the object itself when you call `CheerLeaderList.new`)

Reply:
This 's interesting, I've seen so too , but not sure why !